src/backend/Comments.mo

```diff
@@ -225,4 +233,46 @@ module {
         );
         List.toArray(comments);
     };
+
+    
+    //check principal is admin or not
+    func isAdmin(p : Principal) :  Bool {
+        if(Principal.equal(p , Principal.fromText(ADMIN[0]))){
+            true
+        }else if(Principal.equal(p , Principal.fromText(ADMIN[1]))){
+            true
+        }else {
+            return false
+        }
+    
+    };
+
+    public func deleteComment(state: State, owner: Principal, commentHash: CommentHash) : DeleteResult {
+    // Check if user is an admin
+
+    if (not isAdmin(owner)) {
+        return #err(# NotAdmin)
+    };
+
+    //  argument funtion for list.some
+    func change(x : CommentHash) : Bool {
+    x == commentHash;
+    };
+
+    // Check if the comment exists
+    switch(state.commentStore.get(commentHash)) {
+        case(null) {  return #err(# CommentNotFound)};
+        case(?comment) {
+            let array = List.toArray<CommentHash>(state.commentHistory);
+            func check(arg : CommentHash) : Bool {
+               not (arg == commentHash)
+            };
+            let newA = Array.filter<CommentHash>(array, check);
+        // Delete the comment from the commentStore and update relevant data structures
+          state.commentHistory := List.fromArray<CommentHash>(newA);
+          return #ok()
+         };
+    }; 
```

Comment:
1. Function should also delete the comment from commentStore
2. Should we deduct balance from comment owner?
3. Should reward return to treasury? (I say no)
4. users will have the comment has in their likes. its very expensive to iterate over all users and delete from their like history. Maybe keep it there?